openquake/hazardlib/calc/disagg.py

```diff
@@ -452,9 +453,10 @@ def _disagg6D(self, imldic, g):
         for m, imt in enumerate(imts):
             imlog2[m] = to_distribution_values(iml2[m], imt)
         mea, std = self.mea[self.magi], self.std[self.magi]
+        gp = self.src_mutex.get('grp_probability', 1.)
```

Comment:
In my test case, I see `gp` is being defaulted to 1.0 as the `grp_probability` is instead being stored as `probability` 
`{'src_id': array([ 1, 13], dtype=uint32), 'weight': array([0.5, 0.5]), 'probability': 0.2, 'start': array([0]), 'stop': array([1])}
`
Changing this like to take the `probability` instead, the impact of the `weight` also seems to be missing still